### ISSUE TITLE
Barco G series Projector Support

### ISSUE THREAD

**Issue by GearsHead79** [bitfocus/companion-module-requests#254]
**PLEASE USE THE FOLLOWING PAGE IF YOU WANT SUPPORT FOR A NEW DEVICE OR SOFTWARE PROGRAM**: https://github.com/bitfocus/companion-module-requests/issues/new

Would it be possible to add support for the Barco G Series projectors?
Specifically, the G60 series...

**Comment by josephdadams** [comment]
@GearsHead79 please provide the protocol documentation for this if you want your request to be considered.

**Comment by GearsHead79** [comment]
My apologies!
Here is some information (although I'm not sure if it is what is needed...)

Control: IR, RS232, RJ45, 3.5mm phone jack for wired remote
Network Connection: 10/100 Ethernet, RJ45
(the above is from [https://www.barco.com/en/product/g60-w10#specs.](https://www.barco.com/en/product/g60-w10#specs) Or use [https://www.barco.com/services/website/en/ProductSpecSheet/ProductSpecSheetFile?productId=e52a4eb6-effd-48c0-b76b-098bcb4770ef](https://www.barco.com/services/website/en/ProductSpecSheet/ProductSpecSheetFile?productId=e52a4eb6-effd-48c0-b76b-098bcb4770ef))

Currently, we are using the network connection to the projector with a static IP address. The G60W10 has Crestron Control capabilities, but we neither use that nor do we have it set up in our facility.

Projector Model: G60-W10

Projector Firmware Information (if it is needed):
MCU Version: A01.91
DDP Version: V08.84
M9813 Version: M00.71
LAN Version: B01.14

**Comment by josephdadams** [comment]
What features are you looking to see implemented?

**Comment by bluesteel52502** [comment]
Not OP but could also use this module. Ideal features include:
Lamp control (on/off)
Shutter open/close
Lens Shift
Lens Zoom
Lens Focus

Similar to the modules for Barco DP, HDX, and UDX projectors.

**Comment by bmvideoprod** [comment]
Hi guys, just an update for people looking for an answer:
Add Connection: Generic TCP and UDP Requests
Ip address of the projector Port:3023 (G60-W10)
TCP
Here are some commands: Keep the [...]
Power On [POWR1]
Power Off [POWR0]
Video Mute [PMUT1] (shutter on)
Video Unmute [PMUT0] (shutter off)
Freeze [FREZ1]
Unfreeze [FREZ0]

**Comment by josephdadams** [comment]
No. At this point really the only way new modules get made is if someone who knows how to code wants to make it or if the requester hires someone to make it. Just too much to do otherwise.

**Comment by bmvideoprod** [comment]
I just changed my comment with a solution.

<https://www.linkedin.com/in/bernard-mauran-780851133/>  *Bernard Mauran *
   ***@***.**** ***@***.***>
        +1 (514) 554 7761

**Comment by josephdadams** [comment]
Great!

**Comment by josephdadams** [comment]
Closing as **not planned** during a backlog cleanup of long-standing projector requests.

Nearly all professional projectors from this manufacturer implement **PJLink**, and Companion ships the [generic-pjlink](https://bitfocus.io/connections/generic-pjlink) module, which handles power on/off, input selection, shutter/AV mute, and lamp/error status. That covers the common use case for a Stream Deck without needing a model-specific module. Barco also has [barco-hdx](https://bitfocus.io/connections/barco-hdx), [barco-pulse](https://bitfocus.io/connections/barco-pulse) and [barco-dp](https://bitfocus.io/connections/barco-dp), which may fit this model.

This request predates 2023 and hasn't seen movement, so I'm closing it rather than leaving it sitting. If PJLink genuinely doesn't cover what you need — lens control, warping, edge blend, or vendor-specific presets — please comment with the specific functions and a link to the protocol documentation, and we'll reopen.
